### PR TITLE
3-Validations-and-model-specs

### PR DESCRIPTION
**Pull request by yuvenalmash**
## Changes Summary
- Add validations:
  - User model:
    - Name must not be blank
    - PostsCounter must be an integer greater than or equal to zero
  - Post model:
    - Title must not be blank
    - Title must not exceed 250 characters
    - CommentsCounter must be an integer greater than or equal to zero
    - LikesCounter must be an integer greater than or equal to zero
- Add unit specs for all of your models' methods and validations

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      post = Post.new(title: 'Post 1', author: @user, comments_counter: 0, likes_counter: -1)
+      expect(post).to_not be_valid
+    end
+  end
+
+  describe 'recent_comments' do
+    it 'returns the 5 most recent comments' do
+      user = User.create(name: 'User 1')
+      user.posts.create(title: 'Post 1', text: 'Post 1 text')
+      post.comments.create(text: 'Comment 1', author: user) # comment1
```
Comment (elisha2kyakpo1) [inline spec/models/post_spec.rb:50]: - [x] Almost all the tests passed, but one test failed. I think you need to define a variable post before using it. I have attached an image for your reference.

post

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @yuvenalmash, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Your readme file looks descriptive.
👍 All the model methods are covered by the test specs.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (yuvenalmash) [reply spec/models/post_spec.rb:50]: - Thank you, I must have deleted it while fixing the linter errors.

**Review by DeliceLydia — APPROVED** [review]
## Hi @yuvenalmash, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- [ ] Addressed all changes


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**